### PR TITLE
Fix inconsistent argument exit code when argparse exit with its own error code

### PR DESCRIPTION
**Pull request by dmrlawson**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

We were bitten by this causing us to think that pylint was completing successfully when in fact a typo in a configuration file was causing argparse to exit with exit code 2. This didn't happen in older versions of pylint for whatever reason. The [documentation for pylint](https://pylint.pycqa.org/en/latest/user_guide/usage/run.html#exit-codes) suggests that this means "error message issued" but that pylint executed properly when this is not the case.

It feels a bit clumsy to just add in `try..except`, but on the other hand it's perhaps unexpected that argparse will raise a `SystemExit`. I'm open to any suggestions to improve this.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Nice catch, we indeed missed that argparse could exit with its own exit code and the inconsistency with pylint's error codes. I think we should use -1 because it's the only way our bit encoded error code can be extended to 64 if required. Thoughts @DanielNoord ?

**Comment by dmrlawson** [comment]
> Nice catch, we indeed missed that argparse could exit with its own exit code and the inconsistency with pylint's error codes. I think we should use -1 because it's the only way our bit encoded error code can be extended to 64 if required. Thoughts @DanielNoord ?

`sys.exit(32)` appears throughout the code. The "usage error" exit code could be factored out to be a constant and then changed but it's a bit more than I was intending here.

```
$ grep -r 'sys.exit(32)'
pylint/pyreverse/utils.py:        sys.exit(32)
pylint/pyreverse/utils.py:        sys.exit(32)
pylint/lint/run.py:            sys.exit(32)
pylint/lint/run.py:            sys.exit(32)
pylint/lint/run.py:            sys.exit(32)
pylint/lint/run.py:                sys.exit(32)
pylint/config/arguments_manager.py:            sys.exit(32)
pylint/config/config_initialization.py:        sys.exit(32)
pylint/config/config_initialization.py:            sys.exit(32)
```

**Comment by Pierre-Sassoulas** [comment]
You're right let's keep it as 32, I wasn't up to date on this aspect of the code sorry.

**Review by DanielNoord — DISMISSED** [review]

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Could you fix the tests (there's some other test that is somewhat fragile you'll need to add the new file to it) and add a changelog entry please ?

File: doc/whatsnew/fragments/7931.bugfix [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fix exit code when argparse raises SystemExit due to bad arguments.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/fragments/7931.bugfix:1]:
```suggestion
When pylint exit due to bad arguments being provided the exit code will now be the expected ``32``.
```

**Review by Pierre-Sassoulas — DISMISSED** [review]
Great job 👌

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**